### PR TITLE
`Code` `RolePlaying` example

### PR DESCRIPTION
**Pull request by lightaime**
## Description

Add an example of using `RolePlaying` for `Code` task.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Example (update in the folder of example)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)

### CODE REVIEW

File: examples/code/role_playing.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+from camel.utils import print_text_animated
+
+
+def main(model_type=None) -> None:
+    task_prompt = "Develop a poll app"
+    language = "JavaScript"
+    domain = "Sociology"
+    meta_dict = {"language": language, "domain": domain}
+    role_play_session = RolePlaying(
+        f"{language} Programmer",
```
Comment (Obs01ete) [inline examples/code/role_playing.py:27]: I think this in-place prompt engineering should not be here. The cool thing about camel is that it incapsulate all the prompts. So the external API can only enter the parameters and all the prompt engineering happens behind the API. This is for the future rework.

File: examples/code/role_playing.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+from camel.typing import TaskType
+from camel.utils import print_text_animated
+
+
+def main(model_type=None) -> None:
+    task_prompt = "Develop a poll app"
+    language = "JavaScript"
+    domain = "Sociology"
+    meta_dict = {"language": language, "domain": domain}
+    role_play_session = RolePlaying(
```
Comment (Obs01ete) [inline examples/code/role_playing.py:26]: Thanks for the example. I will update the code for Agents accordingly.

**Review by Obs01ete — APPROVED** [review]

File: examples/code/role_playing.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from colorama import Fore
+
+from camel.societies import RolePlaying
+from camel.typing import TaskType
+from camel.utils import print_text_animated
+
+
+def main(model_type=None) -> None:
```
Comment (Obs01ete) [inline examples/code/role_playing.py:21]: Let's also put main into tests/CI since you have it parametrized

**Review by Obs01ete — CHANGES_REQUESTED** [review]

File: examples/code/role_playing.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    domain = "Sociology"
+    meta_dict = {"language": language, "domain": domain}
+    role_play_session = RolePlaying(
+        f"{language} Programmer",
+        f"Person working in {domain}",
+        task_prompt=task_prompt,
+        with_task_specify=True,
+        model_type=model_type,
+        task_type=TaskType.CODE,
+        extend_sys_msg_meta_dicts=[meta_dict, meta_dict],
```
Comment (Obs01ete) [inline examples/code/role_playing.py:33]: This is too complicated as an API. Very unintuitive. We need to segregate prompt engineering of AI society and Code from the dialogue algorithm of RolePlaying. I would create an instance of RolePair into the constructor of RolePlaying. AISocietyRoles and CodeRoles are the concrete classes inherited from RolePair. TDYT @lightaime

Comment (Obs01ete) [reply examples/code/role_playing.py:33]: This is the right place to apply **dependency inversion** to disentangle {AISocietyRoles and CodeRoles} from RolePlaying via the interface RolePair.

File: examples/code/role_playing.py (hunk 1 above)
Comment (lightaime) [reply examples/code/role_playing.py:27]: Sounds good

File: examples/code/role_playing.py (hunk 2 above)
Comment (lightaime) [reply examples/code/role_playing.py:26]: Cool

File: examples/code/role_playing.py (hunk 3 above)
Comment (lightaime) [reply examples/code/role_playing.py:21]: Added

**Review by Obs01ete — APPROVED** [review]
All good

**Pull request merged**